app/javascript/components/public_health/PatientsTable.js

```diff
@@ -91,55 +91,76 @@ class PatientsTable extends React.Component {
   }
 
   componentDidMount() {
-    // load saved tab from local storage if present
+    // load local storage variables when present
+    const query = {};
+
+    // Set tab from local storage if it exists and is a valid tab
     let tab = localStorage.getItem(`${this.props.workflow}Tab`);
-    if (tab === null || !(tab in this.props.tabs)) {
-      tab = this.state.query.tab;
-      localStorage.setItem(`${this.props.workflow}Tab`, tab);
+    if (tab === null || !Object.keys(this.props.tabs).includes(tab)) {
+      query.tab = this.state.query.tab;
+      localStorage.setItem(`${this.props.workflow}Tab`, query.tab);
+    } else {
+      query.tab = tab;
     }
 
-    // select tab and fetch patients
-    this.handleTabSelect(tab);
-
     // Set jurisdiction if it exists in local storage
     let jurisdiction = localStorage.getItem('SaraJurisdiction');
     if (jurisdiction) {
-      this.handleJurisdictionChange(parseInt(jurisdiction));
+      query.jurisdiction = parseInt(jurisdiction);
+    }
+
+    // Set scope if it exists in local storage
+    let scope = localStorage.getItem('SaraScope');
+    if (scope) {
+      query.scope = scope;
     }
 
     // Set assigned user if it exists in local storage
     let assigned_user = localStorage.getItem('SaraAssignedUser');
     if (assigned_user) {
-      this.handleAssignedUserChange(assigned_user);
+      query.user = assigned_user;
     }
 
     // Set search if it exists in local storage
     let search = localStorage.getItem(`SaraSearch`);
     if (search) {
-      this.setState(
-        state => {
-          return {
-            query: { ...state.query, search: search },
-          };
-        },
-        () => {
-          this.updateTable(this.state.query);
-        }
-      );
+      query.search = search;
     }
 
-    // Select page if it exists in local storage
+    // Set page & sort settings if they exist in local storage & user is in the same workflow as before
+    let priorWorkflow = localStorage.getItem(`Workflow`);
     let page = localStorage.getItem(`SaraPage`);
-    if (page) {
-      this.handlePageUpdate(JSON.parse(page));
+    let sortField = localStorage.getItem(`SaraSortField`);
+    let sortDirection = localStorage.getItem(`SaraSortDirection`);
+    if (priorWorkflow && this.props.workflow === priorWorkflow) {
+      if (page) {
+        query.page = parseInt(page);
+      }
+      if (sortField && sortDirection) {
+        query.order = sortField;
+        query.direction = sortDirection;
+      }
+      // Set a flag so any sticky advanced filter does not reset page & sort settings
+      localStorage.setItem('KeepCurrentSettings', true);
```

Comment:
I'd like to highlight this in case there is a better way of handling this situation - When a sticky Advanced Filter is present, applying that filter is handled elsewhere through [advancedFilterUpdate() ](https://github.com/SaraAlert/SaraAlert/pull/685/files?file-filters%5B%5D=.js#diff-605db0c430cb0318fcd9986ece737d10d16c2ab6807a7c35737484322530bb49L346). As a result, I set this flag in local storage so the page & sort sticky settings are not overwritten by the `setState()` call in `advancedFilterUpdate()`. But in the future, if another special case is created in `PatientsTable` where a filter/setting is applied outside of `componentDidMount()`, that might require adding a new flag to local storage since the `advancedFilterUpdate()` function clears this flag when done. 

If anyone has a possible suggestion or idea on a better way of handling this situation, I'm certainly open to thoughts!

Reply:
Is there a situation in which we _wouldn't_ want the advanced filter to use the page and sort sticky settings if they're available? If no, I wouldn't use this flag at all and instead in `advancedFilterUpdate` just check for these fields in local storage.

Reply:
So it seems like the situation in which we wouldn't want the advanced filter to use the page and sort sticky settings (if they're even available) is if there if the workflow has changed from what it was previously, is that an accurate understanding? If that's the case, I don't think we need this flag because you already check for this case now in `componentDidMount` here, and if it's not true you remove those local storage settings. So I think in 
`advancedFilterUpdate()` all it should have to do is check for the presence of those local storage settings (sort and page). 

As a side note, keep in mind the Advanced Filter is also used in Custom Export too so we'll want to check all of these things don't have undesired effects there as well.

Reply:
Sadly, I don't think that just checking for the existence of page setting in localStorage will work in this case (let me know if I've misunderstood what you were trying to convey). That idea does work for when the patient table is initially loaded with the sticky settings. But then, as an example, if the user were to have been on page 4 of the table and update the advanced filter, since the page is in local storage, the table would remain on page 4, but we would actually want it reset to page 1 in this situation. Of course for the case of sort settings, as you mentioned in your later comments Kylie, I don't think it would make sense to clear out sort settings during an advanced filter update anyway so the sort sticky setting will work by simply checking for their existence in localStorage. 

So here is what we need the solution to achieve (1) when the table is reloaded and this function is called when there was a sticky advanced filter in localStorage, the table should still be on the same page number as before and (2) whenever the advanced filter is changed or updated, the table should be set back to page 1.

Let me know your thoughts or if you have any other ideas on how I may be able to account for both of these scenarios in the same solution.

Reply:
Kylie and I discussed this in more detail last week so I'm following up here regarding some digging I did since then. So whenever there is a sticky advanced filter, a call to `advancedFilterUpdate()` is executted after the Patients Table `componentDidMount()` call concludes which was the reason this change I highlighted was made. This particular call to `advancedFilterUpdate()` originates from [line 278](https://github.com/SaraAlert/SaraAlert/blob/master/app/javascript/components/public_health/query/AdvancedFilter.js#L278) of `AdvancedFilter.js` which traces back to this component's own `componentDidMount()` function call (partial stack trace below).

Given this, Kylie and I briefly discussed the possiblity of including an additional boolean argument for the `advancedFilterUpdate()` function which it can use to determine when it should and shouldn't retain the sticky filters/settings. It could be set to true in the AdvancedFilter's `componentDidMount()` eventual call to `advancedFilterUpdate()` and all other calls could default the boolean arguement to false. If this strategy sounds better that the current implementation, I can proceed with making these updates. Let me know your thoughts. 

![image](https://user-images.githubusercontent.com/42656458/107273432-0d059e80-6a1d-11eb-91ca-099abf02140d.png)

Reply:
After talking with Kylie, I am proceeding with updating applicable function calls to take in a boolean flag that will account for this situation. This will be included in my next push to remote.